tools/clang/lib/SPIRV/SpirvBuilder.cpp

```diff
@@ -17,5 +17,100 @@ SpirvBuilder::SpirvBuilder(SpirvContext &ctx)
   module = new (context) SpirvModule;
 }
 
+SpirvFunction *SpirvBuilder::beginFunction(QualType returnType,
+                                           SourceLocation loc,
+                                           llvm::StringRef funcName) {
+  assert(!function && "found nested function");
+
+  // Create the function
+  function = new (context) SpirvFunction(
+      returnType, /*id*/ 0, spv::FunctionControlMask::MaskNone, loc);
+
+  // Add a debug name for the function
+  auto *debugName = new (context) SpirvName(loc, function, funcName);
```

Comment:
Hmm, I feel we should have a `name` field baked into `SpirvInstruction` directly and provide API to set there.